### PR TITLE
Add recipe for pyotp

### PR DESCRIPTION
**Pull request by smithsp**

### CODE REVIEW

**Comment by smithsp** [comment]
@kalling Can you look into why this is failing?

**Comment by kalling** [comment]
@smithsp Looks like I should just restrict the recipe to macos and Linux, not win64?

**Comment by smithsp** [comment]
@kalling I didn't understand the error of azure's win_64 failure, but your suggestion seems like a reasonable workaround.

**Comment by kalling** [comment]
@smithsp Ok, skipping windows resulted in mac not building.  Hmm.

The thing it failed on doesn't make a lot of sense, since it appears to be before it even really did anything.
``` 
++ mkdir -p /etc/conda/activate.d
mkdir: /etc/conda/activate.d: Permission denied
```

**Comment by kalling** [comment]
@smithsp Ok, now it's identical to how it was before, but now it's failing on mac in addition to windows.

I'll try figuring this out some more tomorrow.

File: recipes/pyotp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+about:
+  home: https://github.com/pyotp/pyotp
+  license: BSD
+  license_family: BSD
+  license_file: ''
+  summary: Python One Time Password Library
+  description: "PyOTP - The Python One-Time Password Library. PyOTP is a Python library for generating and verifying one-time passwords. It can be used to\
+    \ implement two-factor (2FA) or multi-factor (MFA) authentication methods in web applications and in other systems that require users to log in. "
+  doc_url: 'https://github.com/pyotp/pyotp/blob/master/Changes.rst'
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]:
```suggestion
  doc_url: https://pyotp.readthedocs.io/
```

File: recipes/pyotp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+test:
+  imports:
+    - pyotp
+
+about:
+  home: https://github.com/pyotp/pyotp
+  license: BSD
+  license_family: BSD
+  license_file: ''
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]:
```suggestion
  license_file: LICENSE
```

File: recipes/pyotp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - python
+
+test:
+  imports:
+    - pyotp
+
+about:
+  home: https://github.com/pyotp/pyotp
+  license: BSD
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]: This should state [which variant of the BSD licence it is](https://opensource.org/licenses/alphabetical) like `BSD-2-Clause` or `BSD-3-Clause`.

File: recipes/pyotp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    - pip
+  run:
+    - python
+
+test:
+  imports:
+    - pyotp
+
+about:
+  home: https://github.com/pyotp/pyotp
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]:
```suggestion
  home: https://github.com/pyauth/pyotp
```

File: recipes/pyotp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  license: BSD
+  license_family: BSD
+  license_file: ''
+  summary: Python One Time Password Library
+  description: "PyOTP - The Python One-Time Password Library. PyOTP is a Python library for generating and verifying one-time passwords. It can be used to\
+    \ implement two-factor (2FA) or multi-factor (MFA) authentication methods in web applications and in other systems that require users to log in. "
+  doc_url: 'https://github.com/pyotp/pyotp/blob/master/Changes.rst'
+  dev_url: 'https://github.com/pyotp/pyotp'
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]:
```suggestion
  dev_url: https://github.com/pyauth/pyotp
```

File: recipes/pyotp/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "dd9130dd91a0340d89a0f06f887dbd76dd07fb95a8886dc4bc401239f2eebd69" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
```
Comment (chrisburr) [inline recipes/pyotp/meta.yaml]: This line isn't needed:
```suggestion
```

File: recipes/pyotp/meta.yaml (hunk 3 above)
Comment (smithsp) [reply recipes/pyotp/meta.yaml]: It was really an MIT license according to https://pypi.org/project/pyotp/

**Comment by smithsp** [comment]
@chrisburr Thanks for the feedback.  I have made the suggested changes, so we just need to wait and see how it builds out.

**Comment by kalling** [comment]
@smithsp Looks like it's merged now, but looking into the windows error, it looks like maybe this is a known issue.  Considering that the mac issue mysteriously went away, I'm guessing that wasn't related to the commit either. 

Found elsewhere.

https://github.com/conda-forge/staged-recipes/issues/7715

**Comment by chrisburr** [comment]
Sorry I forget to comment about those:
 * As you found `win_64` error is a known issue can be ignored.
 * The macOS error was a misconfiguration somewhere that was unrelated to the recipe. It should have been fixed now.

**Pull request merged**